controller/backup_controller.go

```diff
@@ -207,6 +207,19 @@ func getLoggerForBackup(logger logrus.FieldLogger, backup *longhorn.Backup) *log
 	)
 }
 
+func (bc *BackupController) isBackupBeingUsedForVolumeRestore(backupName, backupVolumeName string) error {
+	volumes, err := bc.ds.ListVolumesByBackupVolumeRO(backupVolumeName)
+	if err != nil {
+		return err
+	}
+	for _, v := range volumes {
+		if v.Status.RestoreRequired {
+			return fmt.Errorf("backup %v cannot be deleted due to the ongoing volume %v restoration", backupName, v.Name)
+		}
+	}
+	return nil
+}
```

Comment:
The code is still outdated. It should have checking replica restoring status as well.

Reply:
This is cherry pick, so the following change is not in this PR.